### ISSUE TITLE
Tensorflow re-implementation

### ISSUE THREAD

**Issue by tdeboissiere** [edouardoyallon/pyscatwave#5]
I was curious to see what would happen if one could backprop through the Scattering transform (in the context of GANs for instance) so I ended up  [re-implementing your code in pure TF](https://github.com/tdeboissiere/DeepLearningImplementations/tree/master/ScatteringTransform).

Whenever I used or modified your work, I added a link to your repo. Is this enough to respect the clauses of the BSD license or should I add an explicit copyright notice ?

My eventual Hybrid GANs were not very successful unfortunately...

**Comment by edouardoyallon** [comment]
Hi, woaw, great job! The explicit copyright would be welcome, especially on scattering.py & cuda routines..

**Comment by szagoruyko** [comment]
@tdeboissiere take a look at https://github.com/edouardoyallon/pyscatwave/pull/6, looks like this bug exists in your version too - sorry about that, we only found it recently.

**Comment by tdeboissiere** [comment]
I have had issues during GAN training when the gradients flow through the Scattering op :  I would  sometimes get abrupt NaNs (even though from tensorboard inspection the weights and gradients look fine).
 
@szagoruyko  could this be related to #6 ?

**Comment by szagoruyko** [comment]
the implementation was incorrect, it was filling imaginary part with the same values as real instead of zeroes, so yes, might cause nans I guess

**Comment by eugenium** [comment]
just to be clear @szagoruyko  means the CPU implementation which you based your tf code on was incorrect, the GPU version should not have a problem.

**Comment by edouardoyallon** [comment]
Just to be precise, the NaNs were due to a tensor that was not initialised in the "cpu mode"(which simply means pytorch based codes instead of custom cuda routines). I'm quite surprised that tf doesn't flow any errors... And by the way, with appropriate custom cuda routines for the gradient, tf & pytorch should have the same speed. Thanks again. 👍

**Comment by szagoruyko** [comment]
@edouardoyallon no that's not true, all it was doing was:
```
norm = sqrt(x*x + y*y)
z = norm + i * norm
```
instead of
```
z = norm + i * 0
```
there was no uninitialized memory

**Comment by edouardoyallon** [comment]
Okay, my bad, then I don't get the NaN, the operations are not specifically ill-conditioned!

**Comment by tdeboissiere** [comment]
Fixed the bug but I still get NaNs. I'll investigate more when I get the time.

**Comment by edouardoyallon** [comment]
bon, to do list later
